asustor_main.c

```diff
@@ -401,25 +450,52 @@ static struct asustor_driver_data asustor_fs6706_driver_data = {
 };
 
 /*
- * It currently looks like the older systems are easier to tell apart, at least if one doesn't insist
- * on detecting the 2 vs 4 vs 6 drives versions (I only did this for AS67xx because I had to do the
- * advanced detection anyway)
+ * AS66xx and AS5304 (maybe all 53xx?) can't be told apart just by DMI either,
+ * (both are "Insyde" "GeminiLake"), so we need to match PCI IDs here as well
  */
+static struct asustor_driver_data asustor_5304_driver_data = {
+	.name = "AS53xx",
+	.pci_matches = {
+		// ASMedia ASM2806 4-Port PCIe x2 Gen3 Packet Switch
+		// (does NOT exist here, but on AS602T and AS604T)
+		{ 0x1b21, 0x2806, 0, 0 }
+	},
+	.leds = &asustor_5304_gpio_leds_lookup,
+	.keys = &asustor_6100_gpio_keys_lookup,
+};
 
 static struct asustor_driver_data asustor_6600_driver_data = {
+	.name = "AS66xx",
+	.pci_matches = {
+		// ASMedia ASM2806 4-Port PCIe x2 Gen3 Packet Switch
+		// (exists on AS602T and AS604T, but not AS5304T)
+		// in the lspci outputs I've seen both had 5 such devices,
+		// but let's be a bit more flexible here..
+		{ 0x1b21, 0x2806, 3, DEVICE_COUNT_MAX }
+	},
 	// NOTE: This is (currently?) the same as for AS6700
 	//       because it seems to use the same GPIO numbers,
 	//       but listed extra for the different name
-	.name = "AS66xx",
-	// This (and the remaining systems) don't need to match PCI devices to be detected,
-	// so they're not set here (and thus initialized to all-zero)
-
 	// the LED GPIOs are the same as in AS67xx, so use the one from AS6704 which should work for
 	// both AS6602T and AS6604T (an AS66xx with more than 4 drives doesn't seem to exist)
 	.leds = &asustor_as6704_gpio_leds_lookup,
 	.keys = &asustor_6100_gpio_keys_lookup,
 };
 
+/*
+ * It currently looks like the older systems are easier to tell apart, at least if one doesn't insist
+ * on detecting the 2 vs 4 vs 6 drives versions (I only did this for AS67xx because I had to do the
+ * advanced detection anyway)
+ */
+
+static struct asustor_driver_data asustor_6300_driver_data = {
+	.name = "AS63xx",
+	// This (and the remaining systems) don't need to match PCI devices to be detected,
```

Comment:
Nit: This comment raises rather than answers a question, please see if you can make it answer "why", or consider if it's useful at all.

Reply:
it explains why, in contrast to the previous driver data definitions, `.pci_matches` is not initialized here or in the remaining ones